core/src/main/java/org/fao/geonet/kernel/EditLib.java

```diff
@@ -282,9 +274,22 @@ private void addChildToParent(MetadataSchema mdSchema, Element targetElement, El
                     LOGGER_ADD_ELEMENT.debug("####		- add child {}", existingChild.toString());
                 }
             }
-            if (qname.equals(singleType))
+            if (qname.equals(singleType)) {
                 targetElement.addContent(childToAdd);
+            }
+
+            filterOnQname(existingAllType, "geonet:child")
+                .stream()
+                .filter(gnChild -> (gnChild.getAttributeValue("prefix") + ":" + gnChild.getAttributeValue("name")).equals(singleType))
+                .findFirst()
+                .ifPresent(targetElement::addContent);
         }
+
+        Stream.concat(
+            filterOnQname(existingAllType, "geonet:element").stream(),
+            filterOnQname(existingAllType, "geonet:attribute").stream()
+        ).forEach(targetElement::addContent);
```

Comment:
Indeed, here the loop is over element types from the schema and was not restoring `geonet:*` in the targetElement. So adding a contact in identification, the targetElement after insertion now contains geonet elements eg. 

```xml
  </gmd:supplementalInformation>
  <geonet:element xmlns:geonet="http://www.fao.org/geonetwork" ref="84" parent="83" uuid="gmd:MD_DataIdentification_58f3d735-4d9f-4d73-b884-0bc29d0be0fc" min="1" max="1" />
  <geonet:attribute xmlns:geonet="http://www.fao.org/geonetwork" name="id" add="true" />
  <geonet:attribute xmlns:geonet="http://www.fao.org/geonetwork" name="uuid" add="true" />
</gmd:MD_DataIdentification>
```